Vice.js

```diff
@@ -148,32 +64,77 @@ function getSearchResults(doc, checkOnly) {
 	return found ? items : false;
 }
 
+function scrape(doc, url) {
+	const trans = Zotero.loadTranslator('web');
+	trans.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48'); // Embedded Metadata
+	trans.setDocument(doc);
 
-function doWeb(doc, url) {
-	switch (detectWeb(doc, url)) {
-		case "multiple":
-			Zotero.selectItems(getSearchResults(doc, false), function (items) {
-				if (!items) return;
+	trans.setHandler('itemDone', function (obj, item) {
+		item.url = url;
+		item.publicationTitle = item.publicationTitle ? item.publicationTitle : "Vice";
+		item.publicationTitle = item.publicationTitle.replace(/^I-d$/, 'i-D');
+		item.ISSN = "1077-6788";
 
-				var articles = [];
-				for (var i in items) {
-					articles.push(i);
-				}
-				ZU.processDocuments(articles, scrape);
-			});
-			break;
-		default:
-			scrape(doc, url);
-			break;
+		let linkedData = text(doc, 'script[type="application/ld+json"]');
+		if (linkedData) {
+			linkedData = JSON.parse(linkedData);
+			if (linkedData && linkedData['@graph']) {
+				linkedData = linkedData['@graph'].find(x => x['@type'] == "NewsArticle");
+				if (linkedData) item.date = linkedData.datePublished;
+			}
+		}
+
+		item.creators.length = 0;
+		for (const author of doc.querySelectorAll('main.main-content .contributors .contributor .contributor__meta a')) {
```

Comment:
Can we make this a bit shorter to be more resilient to site changes? Like:

```suggestion
		for (const author of doc.querySelectorAll('main .contributor__meta a')) {
```

It's less specific, but it still seems to match fine and it'll survive a reorganization of the site's outer layout elements.

Reply:
Good call, it seems to work well, I'll opt for that instead

Reply:
Cool, sounds good.